pkg/storage/chunk/client/openstack/swift_thanos_object_client.go

```diff
@@ -0,0 +1,44 @@
+package openstack
+
+import (
+	"context"
+
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/thanos-io/objstore"
+
+	"github.com/grafana/loki/v3/pkg/storage/bucket"
+	"github.com/grafana/loki/v3/pkg/storage/chunk/client"
+	"github.com/grafana/loki/v3/pkg/storage/chunk/client/hedging"
+)
+
+func NewSwiftThanosObjectClient(ctx context.Context, cfg bucket.Config, component string, logger log.Logger, hedgingCfg hedging.Config) (client.ObjectClient, error) {
```

Comment:
@JoaoBraveCoding can you rebase main? this won't be required as this is already taken care off by bucket.NewObjectClientAdapter()